yt/visualization/_handlers.py

```diff
@@ -0,0 +1,446 @@
+            setattr(self, name, None)
+
+    @property
+    def has_norm(self) -> bool:
+        return self._norm is not None
+
+    def _reset_norm(self):
+        if not self.has_norm:
+            return
+        mylog.warning("Dropping norm (%s)", self.norm)
+        self._norm = None
+
+    def to_float(self, val: un.unyt_quantity) -> float:
+        return float(val.to(self.display_units).d)
+
+    def to_quan(self, val) -> un.unyt_quantity:
+        if isinstance(val, un.unyt_quantity):
+            return self.ds.quan(val)
+        elif (
+            is_sequence(val)
+            and len(val) == 2
+            and isinstance(val[0], Real)
+            and isinstance(val[1], (str, un.Unit))
+        ):
+            return self.ds.quan(*val)
+        elif isinstance(val, Real):
+            return self.ds.quan(val, self.display_units)
+        else:
+            raise TypeError(f"Could not convert {val!r} to unyt_quantity")
+
+    @property
+    def display_units(self) -> un.Unit:
+        return self._display_units
+
+    @display_units.setter
+    def display_units(self, newval: Unit) -> None:
+        self._display_units = un.Unit(newval, registry=self.ds.unit_registry)
+
+    def _set_quan_attr(
+        self, attr: str, newval: Optional[Union[Quantity, float]]
+    ) -> None:
+        if newval is None:
+            setattr(self, attr, None)
+        else:
+            try:
+                quan = self.to_quan(newval)
+            except TypeError as exc:
+                raise TypeError(
+                    "Expected None, a float, or a unyt_quantity, "
+                    f"received {newval} with type {type(newval)}"
+                ) from exc
+            else:
+                setattr(self, attr, quan)
+
+    @property
+    def vmin(self) -> Optional[Union[un.unyt_quantity, Literal["min"]]]:
+        return self._vmin
+
+    @vmin.setter
+    def vmin(self, newval: Optional[Union[Quantity, float, Literal["min"]]]) -> None:
+        self._reset_norm()
+        if newval == "min":
+            self._vmin = "min"
+        else:
+            self._set_quan_attr("_vmin", newval)
+
+    @property
+    def vmax(self) -> Optional[Union[un.unyt_quantity, Literal["max"]]]:
+        return self._vmax
+
+    @vmax.setter
+    def vmax(self, newval: Optional[Union[Quantity, float, Literal["max"]]]) -> None:
+        self._reset_norm()
+        if newval == "max":
+            self._vmax = "max"
+        else:
+            self._set_quan_attr("_vmax", newval)
+
+    @property
+    def dynamic_range(self) -> Optional[float]:
+        return self._dynamic_range
+
+    @dynamic_range.setter
+    def dynamic_range(self, newval: Optional[float]) -> None:
+        if newval is None:
+            return
+
+        try:
+            newval = float(newval)
+        except TypeError:
+            raise TypeError(
+                "Expected a float. " f"Received {newval} with type {type(newval)}"
+            ) from None
+
+        if newval <= 0:
```

Comment:
Checking for 0 is definitely necessary, but strictly negative value should probably work, albeit somewhat counterintuitive to me personally.